Robert Round Robin (#157)
jaybee84/ml-in-rd: content/04.model-complexity.md

### Manage model complexity while preserving the value of machine learning

Translating machine learning findings into testable hypotheses requires the ML models to be both stable – the same predicted features should surface from the data if the model is run multiple times – and simple, as simple models guard against misinterpretation. 
Meeting these requirements is challenging in rare disease datasets where label-noise is abundant. 
In this section we highlight a few common ML techniques that can help improve the stability and simplicity of ML models applied to rare disease data.

Techniques like resampling and combining various ML methods together (ensemble learning) can help achieve stability in predictions (Figure[@fig:2]A-B). 
Resampling without replacement can generate confidence intervals for the model predictions by iteratively exposing the models to incomplete datasets, mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].

> Alternatively, resampling with replacement ('bootstrapping') helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).

Reviewer: ```suggestion
Alternatively, resampling with replacement ("bootstrapping") helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).
```

> Alternatively, resampling with replacement ('bootstrapping') helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).

Author: ```suggestion
Alternatively, resampling with replacement (bootstrapping) helps estimate population values from datasets of limited size, and is also commonly used to find robust models when multiple models are combined into an ensemble ([@doi:10.1080/01621459.1997.10474007; @https://doi.org/10.1023/A:1010933404324; @doi:10.1198/0003130043277; @doi:10/c8xpqz; @doi:10.1016/j.neucom.2004.11.017; @doi:10.1002/sim.4780111607]).
```

> Ensemble learning methods like random forests use _bagging_ (bootstrap aggregation) of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features [@https://doi.org/10.1023/A:1010933404324; @doi:10.1186/1472-6947-13-134; @doi:10.1214/aos/1031689014; @doi:10.1177/2045894019890549; @doi:10/btzfh6]. 

Reviewer: Why does bootstrapping get quotes but bootstrap aggregation does not? Same question about ensemble learning.

Author: Upon reflection I think I prefer it with no quotes, so I changed this for bootstrapping. Maybe need to scrub the rest of the paper for this too....

Regularization simplifies models by making the feature space proportionate with the sample space. (Figure[@fig:2]C)
Regularization can not only protect ML models from poor generalizability caused by overfitting (where the model performs well on held-out training data but poorly on new test data) [@doi:10.1073/pnas.1900654116], but also reduces  model complexity and the feature space to build simpler models. 
Three popular regularized methods, ridge regression, LASSO regression, and elastic-net regression, differ predominantly in how they modify features of the input data. 
Ridge regression can minimize the magnitude of the features, but cannot entirely remove features. 
LASSO regression, on the other hand, works well for selecting a few important features since it can minimize the magnitude of some features more than the others [@doi:10.1038/nmeth.4014]. 
A combination of LASSO and ridge, elastic-net regression [@doi:10.1111/j.1467-9868.2005.00503.x] selects the most useful features, especially in presence of a large number of correlated features.

> Rare variant discovery and immune cell signature discovery studies, like rare diseases, face challenges of the sparsity of observations, and may be a useful model for examining the utility of regularization in scenarios with limited signal.

Reviewer: ```suggestion
Rare variant discovery and immune cell signature discovery studies, like rare diseases, face challenges of the sparsity of observations, and may be useful models for examining the utility of regularization in scenarios with limited signal.
```
